mlflow/utils/requirements_utils.py

```diff
@@ -198,24 +199,29 @@ def _prune_packages(packages):
     return packages - requires
 
 
-def _run_command(cmd):
+def _run_command(cmd, timeout_seconds, env=None):
     """
     Runs the specified command. If it exits with non-zero status, `MlflowException` is raised.
     """
-    proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    stdout, stderr = proc.communicate()
-    stdout = stdout.decode("utf-8")
-    stderr = stderr.decode("utf-8")
-    if proc.returncode != 0:
-        msg = "\n".join(
-            [
-                f"Encountered an unexpected error while running {cmd}",
-                f"exit status: {proc.returncode}",
-                f"stdout: {stdout}",
-                f"stderr: {stderr}",
-            ]
-        )
-        raise MlflowException(msg)
+    proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=env)
+    timer = Timer(timeout_seconds, proc.kill)
+    try:
+        timer.start()
+        stdout, stderr = proc.communicate()
+        stdout = stdout.decode("utf-8")
+        stderr = stderr.decode("utf-8")
+        if proc.returncode != 0:
+            msg = "\n".join(
+                [
+                    f"Encountered an unexpected error while running {cmd}",
+                    f"exit status: {proc.returncode}",
+                    f"stdout: {stdout}",
+                    f"stderr: {stderr}",
+                ]
+            )
+            raise MlflowException(msg)
+    finally:
+        timer.cancel()
```

Comment:
To confirm, can a timer be canceled after it expires?

Reply:
excellent catch :) Updating!